src/Core/Geometry/MeshPrimitives.cpp

```diff
@@ -113,13 +121,17 @@ TriangleMesh makeSharpBox( const Aabb& aabb ) {
         Vector3ui( 20, 21, 22 ), Vector3ui( 20, 22, 23 )  // Top
     };
 
+    if ( bool( color ) )
+        result.colorize( *color );
     result.checkConsistency();
+
     return result;
 }
 
-TriangleMesh makeGeodesicSphere( Scalar radius, uint numSubdiv ) {
+TriangleMesh makeGeodesicSphere( Scalar radius, uint numSubdiv,
+                                 const Utils::optional<Utils::Color>& color ) {
     TriangleMesh result;
-    uint faceCount = std::pow( 4, numSubdiv ) * 20;
+    uint faceCount = uint( std::pow( 4, numSubdiv ) ) * 20;
```

Comment:
This one do not generate warning on my  gcc version 8.3.0 (Debian 8.3.0-6) 
without a cast, but maybe needed by other compiler.
`uint faceCount = std::pow( 4, numSubdiv ) * 20;` compiles fine
(btw uint is not standard right ?)

Reply:
As `pow` returns a floating point value, we actually need a cast to avoid a warning for implicit conversion:
 - `warning: implicit conversion turns floating-point number into integer: 'double' to 'uint' (aka 'unsigned int')`
 - https://en.cppreference.com/w/cpp/numeric/math/pow

Reply:
And yes, uint is not standard, yet massively used in the code.
It is defined here: https://github.com/STORM-IRIT/Radium-Engine/blob/master-v1/src/Core/CoreMacros.hpp#L249